src/Servers/Kestrel/Core/src/Internal/Infrastructure/KestrelEventSource.cs

```diff
@@ -55,16 +55,19 @@ private KestrelEventSource()
     [NonEvent]
     public void ConnectionStart(BaseConnectionContext connection)
     {
-        // avoid allocating strings unless this event source is enabled
```

Comment:
Are the changes in this file important for metrics?  They look like (minor?) perf win, assuming `IsEnabled()` is faster than (uncontested?) `Interlocked`.

Reply:
I put `Interlocked` behind `IsEnabled` checks because it causes performance issues on ARM64 Linux.

Reply:
This is incorrect if you collect counters midway through the process running right? Keeping track of the count and publishing the values are decoupled. This doesn't look correct.

Reply:
Event counters are tracked from the point you start listening to them. All the event source counters, including `Microsoft.AspNetCore.Hosting` counters, work like this.

Metrics counters will be like this as well. If no one listens, the values and tags aren't cached anywhere. Only when someone starts listening do measurements get used.

Reply:
That’s not correct. This behavior was the way it was for a reason, to correctly track the numbers no matter when listening happened. It should continue to do that.